active/TreeView/TreeViewSelectionAPI.md

```diff
@@ -0,0 +1,86 @@
+# Background
+
+The Xaml [TreeView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.TreeView) control shows a list of hierarchical data, similar to a ListView. 
+
+When TreeView was first released, you populated it using the [RootNode property](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.TreeView.RootNodes), giving it a tree of nodes, each node referencing a data item. There was negative feedback about this nodes approach, though, from developers preferring a more data binding friendly pattern along the lines of [ListView.ItemsSource](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ItemsControl.ItemsSource). 
+
+So a [TreeView.ItemsSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.TreeView.ItemsSource) property was added, and now you can use either the nodes approach or the ItemsSource approach. But unlike ListView, TreeView doesn't have the corresponding SelectedItem and SelectedItems properties for single- and multi-select. Those are being added in this spec.
+
+Also being added in this spec, for the case where the RootNode is being used instead of ItemsSource, is an API to select a single node. That is, there is a [SelectedNodes](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.TreeView.SelectedNodes) property today, and this spec is adding a singular SelectedNode property.
+
+Related GitHub issues:  
+https://github.com/microsoft/microsoft-ui-xaml/issues/197  
+https://github.com/microsoft/microsoft-ui-xaml/issues/124  
+https://github.com/microsoft/microsoft-ui-xaml/issues/386
+
+# Description
+
+Extend TreeView selection API sets to make selections easier.
+
+- Add `SelectedNode`: get/set the selected TreeViewNode for single selection
+- Add `SelectedItem`: get/set the selected item for single selection
+- Add `SelectedItems`: get the SelectedItems collection.

```

Comment:
What is the reason that SelectedItems cannot be set via binding? As SelectedItem can be set by two-way binding, it seems to make sense to me to give the same functionality to SelectedItems as well.

`<TreeView ItemsSource="{x:Bind Descendants}" SelectedItems="{x:Bind CurrentDescendants, Mode=TwoWay}" />`

Reply:
This matches [ListView.SelectedItems](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ListViewBase.SelectedItems), but I realize that doesn't answer the question. ListView also supports source collections that implement [ISelectionInfo](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Data.ISelectionInfo), which is a way for the source to have full control over the selection, but it's pretty advanced.

Reply:
@kaiguo @MikeHillberg Yeah, I'd say selecting a group of items should be as simple as possible. Right now, looking at the TreeView Selection API, I need to interact with the specific `TreeView` object to select multiple items. Hence, if I just wanted to mark a group of items in a TreeView as selected, I can't easily do that without somehow referencing an object of the View (as in MVVM). That seems to me like a pretty big oversight and an unnecessary hurdle for the developer.